syntax = "proto3";

package user;

// 用户信息请求
message GetUserInfoRequest {
  int32 user_id = 1;
}

// 用户信息响应
message GetUserInfoResponse {
  bool success = 1;
  string message = 2;
  UserInfo user_info = 3;
  int32 status_code = 4;
  int64 timestamp = 5;
}

// 用户信息
message UserInfo {
  int32 user_id = 1;
  string username = 2;
  string email = 3;
  string status = 4;
  string created_at = 5;
  string last_login = 6;
  string full_name = 7;
  string phone = 8;
  string avatar_url = 9;
}

// 用户服务
service UserService {
  // 获取用户信息
  rpc GetUserInfo(GetUserInfoRequest) returns (GetUserInfoResponse);
} 